feat(skore): Ensure compatibility with pandas 3.0.0

### Describe the bug

Some unit tests fail with `pandas==3.0.0` on my machine.

If I do `python -m pip install "pandas<3"` (love), then all tests pass.


### Steps/Code to Reproduce

Run the tests

### Expected Behavior

No errors

### Actual Behavior

I get this error from a couple of places:

```
>       assert_attr_equal("dtype", left, right, obj=obj)
E       AssertionError: DataFrame.index are different
E       
E       Attribute "dtype" are different
E       [left]:  <StringDtype(na_value=nan)>
E       [right]: object
```

This one for instance:
`FAILED skore/tests/unit/reports/comparison/estimator/metrics/test_numeric.py::test_binary_classification[log_loss-expected5-X_y]`

### Environment

```shell
System:
    python: 3.12.11 (main, Aug 18 2025, 19:19:11) [Clang 20.1.4 ]
executable: /home/arthur/probabl/skore/.venv/bin/python
   machine: Linux-6.17.0-14-generic-x86_64-with-glibc2.39

Python dependencies:
        skore: 0.0.0+unknown
          pip: 26.0.1
    anywidget: 0.9.21
      ipython: 9.10.0
   ipywidgets: 8.1.8
       jinja2: 3.1.6
       joblib: 1.5.3
   matplotlib: 3.10.8
        numpy: 2.4.2
       pandas: 3.0.0
       plotly: 6.5.2
rich[jupyter]: None
 scikit-learn: 1.8.0
      seaborn: 0.13.2
 skore[local]: None
        skrub: 0.7.2
skore-hub-project: 0.0.16
```

--- a/skore/src/skore/_utils/_index.py
+++ b/skore/src/skore/_utils/_index.py
@@ -25,7 +25,7 @@ def flatten_multi_index(index: pd.MultiIndex) -> pd.Index:
     ...     [('a', ''), ('b', '2')], names=['letter', 'number']
     ... )
     >>> flatten_multi_index(mi)
-    Index(['a', 'b_2'], dtype='object')
+    Index(['a', 'b_2'], dtype=...)
     """
     if not isinstance(index, pd.MultiIndex):
         raise ValueError("`index` must be a MultiIndex.")

--- a/skore/tests/unit/displays/table_report/test_common.py
+++ b/skore/tests/unit/displays/table_report/test_common.py
@@ -40,7 +40,7 @@ def display(request):
     return report.data.analyze()
 
 
-@pytest.mark.parametrize("dtype", ["category", "object"])
+@pytest.mark.parametrize("dtype", ["category", "str"])
 @pytest.mark.parametrize("other_label", ["other", "xxx"])
 def test_truncate_top_k_categories(dtype, other_label):
     """Check the behaviour of `_truncate_top_k_categories` when `col` is a categorical

--- a/skore/tests/unit/reports/comparison/estimator/metrics/test_numeric.py
+++ b/skore/tests/unit/reports/comparison/estimator/metrics/test_numeric.py
@@ -19,7 +19,7 @@ from skore import ComparisonReport, EstimatorReport
                     ["DummyClassifier_1", "DummyClassifier_2"],
                     name="Estimator",
                 ),
-                index=pd.Index(["Accuracy"], dtype="object", name="Metric"),
+                index=pd.Index(["Accuracy"], name="Metric"),
             ),
         ),
         (
@@ -64,7 +64,7 @@ from skore import ComparisonReport, EstimatorReport
                     ["DummyClassifier_1", "DummyClassifier_2"],
                     name="Estimator",
                 ),
-                index=pd.Index(["Brier score"], dtype="object", name="Metric"),
+                index=pd.Index(["Brier score"], name="Metric"),
             ),
         ),
         (
@@ -75,7 +75,7 @@ from skore import ComparisonReport, EstimatorReport
                     ["DummyClassifier_1", "DummyClassifier_2"],
                     name="Estimator",
                 ),
-                index=pd.Index(["ROC AUC"], dtype="object", name="Metric"),
+                index=pd.Index(["ROC AUC"], name="Metric"),
             ),
         ),
         (
@@ -86,7 +86,7 @@ from skore import ComparisonReport, EstimatorReport
                     ["DummyClassifier_1", "DummyClassifier_2"],
                     name="Estimator",
                 ),
-                index=pd.Index(["Log loss"], dtype="object", name="Metric"),
+                index=pd.Index(["Log loss"], name="Metric"),
             ),
         ),
     ],
@@ -133,7 +133,7 @@ def test_binary_classification(
                     ["DummyRegressor_1", "DummyRegressor_2"],
                     name="Estimator",
                 ),
-                index=pd.Index(["RMSE"], dtype="object", name="Metric"),
+                index=pd.Index(["RMSE"], name="Metric"),
             ),
         ),
         (
@@ -144,7 +144,7 @@ def test_binary_classification(
                     ["DummyRegressor_1", "DummyRegressor_2"],
                     name="Estimator",
                 ),
-                index=pd.Index(["R²"], dtype="object", name="Metric"),
+                index=pd.Index(["R²"], name="Metric"),
             ),
         ),
     ],

